core/trino-main/src/main/java/io/trino/cost/FilterStatsCalculator.java

```diff
@@ -105,6 +114,31 @@ public PlanNodeStatsEstimate filterStats(
                 .process(simplifiedExpression);
     }
 
+    @VisibleForTesting
+    Optional<Symbol> getSymbolReference(Expression expression, Session session, TypeProvider types)
+    {
+        if (expression instanceof SymbolReference) {
+            return Optional.of(Symbol.from(expression));
+        }
+        if (!(expression instanceof Cast)) {
+            return Optional.empty();
+        }
+        Cast castExpression = (Cast) expression;
+        if (!(castExpression.getExpression() instanceof SymbolReference)) {
+            return Optional.empty();
+        }
+        Map<NodeRef<Expression>, Type> expressionTypes = typeAnalyzer.getTypes(session, types, expression);
+        Type castSourceType = expressionTypes.get(NodeRef.of(castExpression.getExpression()));
+        Type castTargetType = expressionTypes.get(NodeRef.<Expression>of(castExpression));
+        // CAST must be an implicit coercion and the stats representation of target type
```

Comment:
> stats representation

You mean `SymbolStatsEstimate`?

> stats representation of target type should be compatible with the source type

I would say that target domain must be subset of source domain (e.g. `double -> bigint` cast). Is that the case?

Reply:
> > stats representation
> 
> You mean `SymbolStatsEstimate`?

yes; precisely the `io.trino.spi.statistics.StatsUtil#toStatsRepresentation`